### DISCUSSION TITLE
ARM Installation Very Slow

### DISCUSSION THREAD

**Discussion by mbowiewilson** [explosion/spaCy#11006] [Help: Installation] [answered]
Hello 👋  I am a big fan of spaCy, but I am finding that installing spaCy on `linux/arm64` platforms (in Docker) is extremely slow. In contrast, installing spaCy on other platforms such as `linux/amd64` is very fast. The context is I am trying to upgrade my AWS EC2 instances to newer models with the ARM-based Graviton processors.

Here is an minimal Dockerfile from which this issue can be reproduced:
```
FROM python:3.10.4-slim-bullseye
RUN apt-get update && apt-get -y install g++
RUN pip3 install --upgrade pip setuptools wheel && \
    pip3 install spacy
```
and here is the command I am running that takes a very long time: 
```
docker build --platform linux/arm64 -t this_image_builds_slowly -f Dockerfile .
```
The command above takes over 30 minutes, but I don't know exactly how long because I've not had the patience for it to complete.

Any advice would be appreciated. Thanks in advance.

**Comment by pmbaumgartner** [comment, marked as answer]
Hey @mbowiewilson,

I can confirm the slow install time - that step took me `370s` on my M1 Mac. The reason is that we don't yet provide wheels for `linux/arm64`, so it has to build spaCy when it gets installed. 

We're currently working on providing wheels for `linux/arm64`, but I can't provide an ETA on when we'll have them ready.

**Reply by mbowiewilson** [reply]
Thanks for the prompt response @pmbaumgartner! It is interesting that that step took 370 seconds on your M1 Mac. I have a 2019 MacBook Pro with an intel processor and it was taking much, much longer than that for me. Anyway, glad to hear that work to improve this situation is currently in progress :)

**Reply by adrianeboyd** [reply]
As an update, we've just published linux aarch64 wheels on pypi.

**Comment by jaymegordo** [comment]
Still doesn't look like spacy has linux/arm64 wheels on pypi, is this still being worked on?
